testplugins/src/main/java/org/spongepowered/test/ElytraDataTest.java

```diff
@@ -0,0 +1,96 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.test;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandException;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.command.spec.CommandSpec;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.game.state.GamePreInitializationEvent;
+import org.spongepowered.api.plugin.Plugin;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.format.TextColors;
+
+@Plugin(id = "elytratest", name = "ElytraDataTest", description = "Testing some nice elytra data.")
+public class ElytraDataTest {
+
+    @Listener
+    public void onPreInit(final GamePreInitializationEvent event) {
+        final CommandSpec isFlying = CommandSpec.builder()
+            .executor((src, args) -> {
+                if (!(src instanceof Player)) {
+                    throw new CommandException(Text.of(TextColors.RED, "You must be a player to execute this command!"));
+                }
+
+                final Player player = (Player) src;
+
+                player.sendMessage(Text.of(TextColors.GOLD, "Elytra flying state: ",
+                    TextColors.GRAY, player.get(Keys.IS_ELYTRA_FLYING).orElse(false)));
+
+                return CommandResult.success();
+            }).build();
+
+        final CommandSpec setFlying = CommandSpec.builder()
+            .arguments(GenericArguments.onlyOne(GenericArguments.bool(Text.of("enable"))))
+            .executor((src, args) -> {
+                if (!(src instanceof Player)) {
+                    throw new CommandException(Text.of(TextColors.RED, "You must be a player to execute this command!"));
+                }
+
+                final Player player = (Player) src;
+
+                player.offer(Keys.IS_ELYTRA_FLYING, args.<Boolean>getOne("enable").orElse(false));
+
+                player.sendMessage(Text.of(TextColors.GOLD, "You have successfully changed your elytra flying state!"));
+
+                return CommandResult.success();
+            }).build();
+
+        final CommandSpec enable = CommandSpec.builder()
+            .arguments()
+            .executor((src, args) -> {
+                src.sendMessage(Text.of(TextColors.DARK_GREEN, "You have enabled elytra listening."));
+                Keys.IS_ELYTRA_FLYING.registerEvent(Player.class, e -> {
+                    if (e.getTargetHolder() instanceof Player) {
+                        ((Player) e.getTargetHolder()).sendMessage(Text.of(TextColors.DARK_GREEN, "Changed elytra status!"));
```

Comment:
@gabizou So I was working on adding support for the listening to the change of the key, but I couldn't get it to work for manually setting the key or changing elytra state in game. Am I missing a step necessary for implementing the event?

Reply:
It might not be possible to change for the *client* since this is a server side modification, may have to perform it on another player to test properly (Sponge human anyone?)

Reply:
I mean I can change the player to elytra just fine, but the listener is never called. I'm not sure how to properly implement the key change listener.